### DISCUSSION TITLE
kiali 只能关联egg的启动日志

### DISCUSSION THREAD

**Discussion by yangliguo7** [eggjs/egg#5082] [Q&A]
公司日志使用的是kiali，但是关联到egg服务时，只能查看到egg的启动日志

我们本地打得日志，却无法关联，每次都需要进入服务器看日志文件里的日志。java那边日志是可以正常再kiali上查看

请问下是什么地方没设置嘛

```
日志输出方式

    const {
      app: { logger },
    } = this;

logger.error('......');
```

**Comment by atian25** [comment]
看你的 kiali 的日志规范呗， egg 这边只管写到日志文件。
这种一般是日志采集侧有配置规则的，如果他们真的不支持（应该不会那么挫），才需要去定义日志文件规则（看 egg-logger）

**Reply by yangliguo7** [reply]
感谢你的回复。

 kiali是别的部门负责的，所以这边我也不清楚kiali的采集规范是什么，但是egg的启动日志kiali 是能采集到的。

请问下egg启动日志的输出和正常我们使用logger去输出日志的逻辑是否一样呢 ?

**Reply by yangliguo7** [reply]
hi  我debugger看了下代码

在egg启动时， `egg-cluster`里的日志，kiali日志是正常能采集到的

https://github.com/eggjs/egg-cluster/blob/5a656e471d03ec7100fe963836b258b4770e9e54/lib/master.js#L57
```
this.log(`[master] =================== ${frameworkPkg.name} start =====================`);
this.logger.info(`[master] node version ${process.version}`);
```

我原来以为我自己的日志没被采集是启动日志和运行时的日志打印方法不一样，但是我debugger看了下整个egg代码，发现日志打印用的都是统一 `egg-logger` 方法
https://github.com/eggjs/egg-logger/blob/951fab300e44f2534ec6adb633bdcb5858305bb4/lib/logger.js#L63

没看出来啥问题 。[ 苦笑 ]
